service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -119,13 +102,14 @@ impl GenerateProtocol for RestJsonGenerator {
         writeln!(writer, "use serde_json;")?;
 
         // avoid unused imports when building services that don't use params
-        if service.service_type_name() != "Batch" {
+        if service_has_query_parameters(service) {
```

Comment:
Ah, that makes sense. Trait methods can be used like struct methods, just by invoking them as static methods and passing in the object as the first param. For example, here it could be `rusoto_core::param::ServiceParams::put(&mut params, "key", "val")`. Since this code is all generated anyways, I don't think that's _terrible_...